src/Native/build-native.proj

```diff
@@ -0,0 +1,68 @@
+<Project>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), Directory.Build.props))\Directory.Build.props" />
+
+  <!-- Target that builds all the native binaries in the Native folder -->
+  <Target Name="Build" DependsOnTargets="BuildNativeUnix;BuildNativeWindows" />
+
+  <PropertyGroup Condition="'$(OS)' != 'Windows_NT'">
+    <GenerateVersionSourceFile>true</GenerateVersionSourceFile>
+    <NativeVersionSourceFile>$(BaseIntermediateOutputPath)version.c</NativeVersionSourceFile>
```

Comment:
Just to make sure, should this be in the projects intermediate folder or directly under bin/obj/? If the latter then you want to use `RootIntermediateOutputPath`.

Reply:
Then you definitely want to use `RootIntermediateOutputPath` as `BaseIntermediateOutputPath` is project specific.